pkg/controller/sshd/sshd_controller.go

```diff
@@ -355,7 +395,41 @@ func getMatchLabels(cr *cloudingressv1alpha1.SSHD) map[string]string {
 	return map[string]string{"deployment": cr.Name}
 }
 
-func newSSHDDeployment(cr *cloudingressv1alpha1.SSHD, configMapList *corev1.ConfigMapList) *appsv1.Deployment {
+func newSSHDSecret(namespace, name string) (*corev1.Secret, error) {
+	// Generate 2048-bit RSA key
+	rsaKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return nil, err
+	}
+	ssh_host_rsa_key := pem.EncodeToMemory(&pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(rsaKey),
+	})
```

Comment:
@jaybeeunix Security 👀 ?

Reply:
Note, 2048 bits matches [ssh-keygen](https://linux.die.net/man/1/ssh-keygen)'s default for RSA.
> -b bits
Specifies the number of bits in the key to create. For RSA keys, the minimum size is 768 bits and the default is 2048 bits. Generally, 2048 bits is considered sufficient. DSA keys must be exactly 1024 bits as specified by FIPS 186-2.

Reply:
Note, 2048 bits matches [ssh-keygen](https://linux.die.net/man/1/ssh-keygen)'s default for RSA.
> -b bits
Specifies the number of bits in the key to create. For RSA keys, the minimum size is 768 bits and the default is 2048 bits. Generally, 2048 bits is considered sufficient. DSA keys must be exactly 1024 bits as specified by FIPS 186-2.

Reply:
for ref, nist tends to recommend >= 3072 rsa for security needs through 2030, and my openssh on f31 (openssh-8.1p1-1.fc31.x86_64) defaults to 3072 (from the same manpage you mentioned).

it looks like sre-p have said we want 4096 rsa keys: https://github.com/openshift/ops-sop/blob/master/v3/operationsorientation.asciidoc#62-ssh-keys , so let's go with 4096.